jolt-core/src/poly/dense_mlpoly.rs

```diff
@@ -250,11 +250,58 @@ impl<F: JoltField> DensePolynomial<F> {
         assert_eq!(chis.len(), self.Z.len());
         compute_dotproduct(&self.Z, &chis)
     }
+    pub fn split_eq_evaluate(&self, r: &[F], eq_one: &[F], eq_two: &[F]) -> F {
+        const PARALLEL_THRESHOLD: usize = 16;
+        if r.len() < PARALLEL_THRESHOLD {
+            self.evaluate_split_eq_serial(eq_one, eq_two)
+        } else {
+            self.evaluate_split_eq_parallel(eq_one, eq_two)
+        }
+    }
+    fn evaluate_split_eq_parallel(&self, eq_one: &[F], eq_two: &[F]) -> F {
+        let eval: F = (0..eq_one.len())
+            .into_par_iter()
+            .map(|x1| {
+                if eq_one[x1].is_zero() {
+                    F::zero()
```

Comment:
Can remove this check, `eq_one` won't have any zeros (assuming that we're evaluating on a random point)